src/com/ppm/integration/agilesdk/connector/octane/OctaneRequestIntegration.java

```diff
@@ -1311,6 +1312,74 @@ public List<AgileEntityIdProjectDate> getAgileEntityIDsToCreateInPPM(final Strin
         return ids;
     }
 
+    /** @since 10.0.3 */
+    public boolean supportsPPMRequestToExistingAgileEntitySync() {
+        return true;
+    }
+
+
+    /**
+     * Returns the IDs & Names of the Agile Entities of the given entity type in the given agile project that can be mapped to a PPM Request.
+     *<p>
+     * PPM will first verify that these entities don't already have a PPM Request synched to it before being returned to the PPM Users.
+     *<p>
+     * You should only override this method if {@link #supportsPPMRequestToExistingAgileEntitySync()} returns true. This method will never be called if it returns false.
+     *
+     * @param entityType Entity Type of the Agile Entities to return.
+     * @return The list of ID + Name of agile entities that can be mapped to an existing PPM Request.
+     * @since 10.0.3
+     */
+    public List<AgileEntityIdName> getCandidateEntitiesToSyncWithRequests(final String agileProjectValue, final String entityType,
+                                                                          final ValueSet instanceConfigurationParameters) {
+        List<AgileEntityIdName> ids = new ArrayList<AgileEntityIdName>();
+        ClientPublicAPI client = ClientPublicAPI.getClient(instanceConfigurationParameters);
+        Map spaceAndWorkspace = getSpaceAndWorkspace(client, agileProjectValue, entityType);
+        String spaceId = (String)spaceAndWorkspace.get(SPACE_PLACEHOLDER);
+        List<String> workspaceIds = (List)spaceAndWorkspace.get(WORKSPACE_PLACEHOLDER);
+
+        for (String id : workspaceIds) {
+            List<AgileEntityIdName> entitiesCollection =
+                    getExistingEntities(client, spaceId, id, entityType);
+            ids.addAll(entitiesCollection);
+        }
+        return ids;
+    }
+
+    private List<AgileEntityIdName> getExistingEntities(ClientPublicAPI client, String spaceId, String workSpaceId, String entityType) {
+
+
+        // if it is shared epic,its real workspace id is 500.But it could
+        // navigation to using the mask workspace id
+        String realWorkspaceId = workSpaceId;
+        if (OctaneConstants.SUB_SHARED_EPIC.equals(entityType)) {
+            realWorkspaceId = OctaneConstants.SHARED_EPIC_DEFAULT_WORKSPACE;
+            entityType = OctaneConstants.SUB_TYPE_EPIC;
+        }
+
+        Map<String, Object> queryParams = new HashMap<String, Object>();
+
+        if (OctaneConstants.SUB_TYPE_EPIC.equals(entityType)) {
+            Map<String, FieldInfo> fieldMap = getFieldInfoMap(client, spaceId, workSpaceId, entityType);
+            if (fieldMap.containsKey("epic_level")) {
+                queryParams.put(OctaneConstants.FULL_QUERY,
+                        "!epic_level={logical_name EQ ^list_node.epic_level.local^}");
```

Comment:
epic_level has three options. "option portfolio" means create from PPM/SPM, "option local" means create in octane.  "option null" is default. I am not sure which should be selected for this situation.